File: 프로그래머스/lv2/42578. 의상/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> hm = new HashMap<>();
        int answer = 0;
        for(int i=0; i<clothes.length; i++){
            hm.put(clothes[i][1], hm.getOrDefault(clothes[i][1], 0) +1);
        } 
        if(hm.size() > 1) {
            int count = 1;

            for(String key : hm.keySet()){
                count *= hm.get(key) + 1;
            }

            answer = count - 1;
        }else{
            answer = clothes.length;
        }
        return answer;
    }
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const map = new Map();
    for (const [name, type] of clothes) {
        map.set(type, (map.get(type) || 0) + 1);
    }
    
    let answer = 1;
    for (const [key, value] of map) {
        answer *= value + 1;
    }
    
    return answer - 1;
}